base/ship

```diff
@@ -6,7 +6,10 @@
 # - Any executable (absolute or on PATH) can also be specified as the command
 # - Adds a standard way of getting the docker host IP
 
-set -e
+set -e -x -v
+
+# Try to give mesos time to fetch the logs before the container exits on error
+trap 'echo "Unexpected exit from ship, pausing to allow log collection" && sleep 20' EXIT
```

Comment:
So the idea here is that it seems like if a container starts and dies too
quickly, then something in the mesos stack ends up not getting to the step
where it pulls the logs so they aren't available for debugging why it
died.  I'm not entirely confident this is what is actually happens, but
anecdotal evidence suggests it.

This is a bit of a shot in the dark to try to avoid that.  Once we get past
the ship script and things it calls, then the pause no longer applies since
we exec the shell away.

On Thu, Mar 12, 2015 at 11:24 AM, Noah Meyerhans <notifications@github.com>
wrote:

> In base/ship
> <https://github.com/socrata/shipyard/pull/11#discussion_r26330000>:
>
> > @@ -6,7 +6,10 @@
> >  # - Any executable (absolute or on PATH) can also be specified as the command
> >  # - Adds a standard way of getting the docker host IP
> >
> > -set -e
> > +set -e -x -v
> > +
> > +# Try to give mesos time to fetch the logs before the container exits on error
> > +trap 'echo "Unexpected exit from ship, pausing to allow log collection" && sleep 20' EXIT
>
> What's supposed to happen during this 20 second pause?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/socrata/shipyard/pull/11/files#r26330000>.
>